coaster/db.py

```diff
@@ -1,5 +1,43 @@
 # -*- coding: utf-8 -*-
 
-from flask.ext.sqlalchemy import SQLAlchemy
+from __future__ import absolute_import
+from sqlalchemy import exc
+from flask.ext.sqlalchemy import SQLAlchemy as SQLAlchemyBase, SignallingSession
+
+__all__ = ['SQLAlchemy', 'db']
+
+
+class CoasterSession(SignallingSession):
+    """
+    Custom session that provides additional helper methods.
+    """
+
+    def add_and_commit(self, _instance, **filters):
+        """
+        Add and commit a new instance, rolling back if the commit fails with an IntegrityError.
+
+        :param _instance: Instance to commit
+        :param filters: Filters required to load existing instance from the database
+            in case the commit fails (optional).
+        :return: Instance that is in the database (in case the commit failed, returned only if filters are specified)
+        """
+        self.begin_nested()
+        try:
+            self.add(_instance)
+            self.commit()
+            return _instance
+        except exc.IntegrityError:
+            self.rollback()
+            if filters:
+                return self.query(_instance.__class__).filter_by(**filters).one()
```

Comment:
Consider this:

```python
    class Tweet(BaseMixin, db.Model):
        content = db.Column(db.Unicode(140), nullable=False)
        user_id = db.Column(db.Integer, nullable=False)
   
        # user misses the content parameter
        >> t = Tweet(user_id=1)
        >> self.session().add_and_commit(t)
````
Is this a plausible scenario? In which case, it should raise an `IntegrityError` exception? Right now, this returns `None`.